autoload/fireplace.vim

```diff
@@ -1842,10 +1842,13 @@ function! fireplace#source(symbol) abort
   let info = fireplace#info(a:symbol)
 
   let file = ''
-  if !empty(get(info, 'resource'))
```

Comment:
This is deliberately the highest priority, because `file` can be wrong if we're connected to a remote server or something like a Docker container. What does moving it down accomplish?

Reply:
Moving this down allows for a check of `jar:file:` in `file` first in order to use the provided reference to compressed source code (in my case, openjdk 8).

To remedy this, I've included a commit that will try to get the source using resource first and then check the result before proceeding to use file.

Reply:
That sounds right. But `fireplace#findresource()` does look inside jar files in the class path. Should we also be looking inside `(System/getProperty "java.library.path")`? Would that fix it in this case?

Reply:
I am not sure, my `java.library.path` is `/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib`

The src path that is being picked up by cider is `/usr/lib/jvm/java-8-openjdk/src.zip` which is in `usr/lib` but I'm not sure how we could do a deep search to find it.

Reply:
What is `resource`? I imagine the cider-nrepl source can tell me if I know what I'm starting with.

Reply:
`resource` is `java/lang/Runtime.java` when calling `echo fireplace#info('Runtime/getRuntime')` with the openjdk source code package installed.

file is `jar:file:/usr/lib/jvm/java-8-openjdk/src.zip!/java/lang/Runtime.java` if that helps.

Reply:
`java.base/java/lang/System.java` is the `resource` for `System/getProperty`. So `#findresource()` would just need to get its hands on the path to that `src.zip`.